src/main/java/carpet/mixins/FluidBlock_renewableDeepslateMixin.java

```diff
@@ -0,0 +1,30 @@
+package carpet.mixins;
+
+import carpet.CarpetSettings;
+import net.minecraft.block.BlockState;
+import net.minecraft.block.Blocks;
+import net.minecraft.block.FluidBlock;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.World;
+import net.minecraft.world.WorldAccess;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Shadow;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfoReturnable;
+
+@Mixin(FluidBlock.class)
+public abstract class FluidBlock_renewableDeepslateMixin {
+
+    @Shadow protected abstract void playExtinguishSound(WorldAccess world, BlockPos pos);
+
+    @Inject(method = "receiveNeighborFluids", at = @At(value = "INVOKE",target = "Lnet/minecraft/fluid/FluidState;isStill()Z"), cancellable = true)
+    private void receiveFluidToDeepslate(World world, BlockPos pos, BlockState state, CallbackInfoReturnable<Boolean> cir) {
+        if(CarpetSettings.renewableDeepslate && pos.getY() < 0) {
```

Comment:
You would still want to check for stillness to keep obsidian behaviour
```suggestion
        if(CarpetSettings.renewableDeepslate && !world.getFluidState(pos).isStill() && pos.getY() < 0 && world.) {
```
Another thing is to check world generator settings if deepslate is enabled, or just hardcode it to overworld. (deepslate enabled in ChunkGeneratorSettings and its not that easy to get to it)